packages/react-ui/src/app/builder/builder-hooks.ts

```diff
@@ -223,3 +224,67 @@ function getStateAtPath({
   });
   return targetMap;
 }
+
+const getAllStepsMentions: (state: BuilderState) => MentionTreeNode[] = (
+  state,
+) => {
+  const { selectedStep, flowVersion } = state;
+  if (!selectedStep || !flowVersion || !flowVersion.trigger) {
+    return [];
+  }
+  const step = flowHelper.getStep(flowVersion, selectedStep.stepName);
+  if (!step) {
+    return [];
+  }
+  const path = flowHelper.findPathToStep({
+    stepToFind: step,
+    trigger: flowVersion?.trigger,
+  });
+
+  return path.map((s) => {
+    const stepMentionNode: MentionTreeNode =
+      dataSelector.traverseStepOutputAndReturnMentionTree({
```

Comment:
So this is confusing, because why would you traverse the tree then check if It's undefined? and do special case handling for  that?

a better thinking thougt If it's need test return the testStepNode otherwise traverse tree and return the testStepNoode